File: dart_sample/lib/demos/cupertino/refresh_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

class CupertinoRefreshDemoPage extends StatefulWidget {
  const CupertinoRefreshDemoPage({super.key});

  @override
  State<CupertinoRefreshDemoPage> createState() =>
      _CupertinoRefreshDemoPageState();
}

class _CupertinoRefreshDemoPageState extends State<CupertinoRefreshDemoPage> {
  int _refreshCount = 0;
  String _status = 'Pull down from the top';

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 10,
      children: <Widget>[
        const Text(
          'Cupertino sliver refresh',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Pull the list down past the indicator, then release. '
          'The sliver holds 60 px while refreshing.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Text(
          '$_status · refreshCount=$_refreshCount',
          style: const TextStyle(fontSize: 12, color: Colors.blueGrey),
        ),
        Expanded(
          child: CustomScrollView(
            physics: const BouncingScrollPhysics(
              parent: AlwaysScrollableScrollPhysics(),
            ),
            slivers: <Widget>[
              CupertinoSliverRefreshControl(onRefresh: _handleRefresh),
              SliverFixedExtentList.builder(
                itemCount: 24,
                itemExtent: 54,
                itemBuilder: (BuildContext context, int index) => Container(
                  color: index.isEven ? Colors.white : const Color(0xFFF5F7FA),
                  padding: const EdgeInsets.symmetric(
                    horizontal: 12,
                    vertical: 10,
                  ),
                  child: Text(
                    'Cupertino refresh row #${index + 1}',
                    style: const TextStyle(fontSize: 13, color: Colors.black),
                  ),
                ),
              ),
            ],
          ),
        ),
      ],
    );
  }

  Future<void> _handleRefresh() async {
    if (mounted) setState(() => _status = 'Refreshing');
    await Future<void>.delayed(const Duration(milliseconds: 650));
    if (mounted) {
      setState(() {
        _refreshCount += 1;
        _status = 'Refresh complete';
      });
    }
  }
}

File: src/Sample/Plumix.Sample/Demos/Cupertino/RefreshDemoPage.cs
using System.Threading.Tasks;
using Avalonia;
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.Widgets;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/refresh_demo_page.dart (exact sample parity)

namespace Plumix;

public sealed class CupertinoRefreshDemoPage : StatefulWidget
{
    public override State CreateState()
    {
        return new CupertinoRefreshDemoPageState();
    }
}

internal sealed class CupertinoRefreshDemoPageState : State
{
    private int _refreshCount;
    private string _status = "Pull down from the top";

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 10.0,
            children:
            [
                new Text("Cupertino sliver refresh", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "Pull the list down past the indicator, then release. The sliver holds 60 px while refreshing.",
                    fontSize: 14.0,
                    color: Color.Parse("#8A000000")),
                new Text(
                    $"{_status} · refreshCount={_refreshCount}",
                    fontSize: 12.0,
                    color: Color.Parse("#FF607D8B")),
                new Expanded(
                    child: new CustomScrollView(
                        physics: new BouncingScrollPhysics(parent: new AlwaysScrollableScrollPhysics()),
                        slivers:
                        [
                            new CupertinoSliverRefreshControl(onRefresh: HandleRefresh),
                            SliverFixedExtentList.Builder(
                                childCount: 24,
                                itemExtent: 54.0,
                                addAutomaticKeepAlives: false,
                                itemBuilder: (_, index) => new Container(
                                    color: index % 2 == 0 ? Colors.White : Color.Parse("#FFF5F7FA"),
                                    padding: new Thickness(12.0, 10.0),
                                    child: new Text(
                                        $"Cupertino refresh row #{index + 1}",
                                        fontSize: 13.0,
                                        color: Colors.Black))),
                        ])),
            ]);
    }

    private async Task HandleRefresh()
    {
        if (Mounted)
        {
            SetState(() => _status = "Refreshing");
        }

        await Task.Delay(650);
        if (Mounted)
        {
            SetState(() =>
            {
                _refreshCount++;
                _status = "Refresh complete";
            });
        }
    }
}
